Added new UI button option with examples.

StealthyExpertX opened the pull request:
Added a new UI button option with an example, I also have done the due diligence to update the examples in 3_fixed_function_pipeline.py

Added a new private function _create_button() that adds a new button UI type for the simple plugin UI API.
The first argument is the "button name", and the second argument is the function to call as an event when the button is clicked/pressed by the user.

#custom function you want to run from the button click event.
def function_name():
    print("you have clicked a button.")

#button input type with two arguments.
operation_options = {
    "Button input": ["button", "button name", function_name]
}

StealthyExpertX committed:
Added new UI button option with examples.

Added new UI button option with example, I also have done the due diligence to update the examples in 3_fixed_function_pipeline.py

gentlegiantJGC commented:
A couple of issues.
The `callback` variable is shadowing the outer scope. Try to avoid name shadowing to avoid confusion.
A options value of `[]` will lead to callback being undefined at line 224 and an options value of `["name"]` will lead to an unpacking error on line 217.
Is there benefit to having a button with no callback function?

I think it would make sense to handle unpacking the variables in the actual function definition rather than within the function which will make this a bit simpler.
I will make these changes.

gentlegiantJGC committed:
Simplified and fixed the _create_button method

gentlegiantJGC committed:
Improved fixed function pipeline

Switched to *args as inputs. This will get swapped to positional inputs in the next commits.
Added logging information if something goes wrong.

gentlegiantJGC committed:
Refactored option handling in fixed pipeline

Instead of code in the function handling unpacking the options and filling default types we can use the function itself to unpack and handle filling defaults for optional parameters.
We still need to make sure the types are correct but this is much easier now.

StealthyExpertX commented:
Alright, thanks for the pointers!
